WhaleSpotting/Repositories/WhaleSightingRepo.cs

```diff
@@ -32,4 +33,36 @@ public WhaleSighting GetById(int id)
             throw new ArgumentOutOfRangeException($"No sightning with id {id} found in the database", ex);
         }
     }
+
+    public List<WhaleSighting> Search(WhaleSightingSearchRequest whaleSightingSearchRequest) {
+        try
+        {
+            var query = context.WhaleSightings
+                .Include(ws => ws.User)
+                .Include(ws => ws.WhaleSpecies)
+                .ToList();
+
+                if (!string.IsNullOrEmpty(whaleSightingSearchRequest.WhaleSpecies)) query = query.Where(ws => ws.WhaleSpecies.Name.ToLower() == whaleSightingSearchRequest.WhaleSpecies.ToLower()).ToList();
+                
+                if (!string.IsNullOrEmpty(whaleSightingSearchRequest.Colour)) query = query.Where(ws => ws.WhaleSpecies.Colour.ToLower() == whaleSightingSearchRequest.Colour.ToLower()).ToList();
+
+                if (whaleSightingSearchRequest.TailType != null) query = query.Where(ws => ws.WhaleSpecies.TailType == whaleSightingSearchRequest.TailType).ToList();
+
+                if (whaleSightingSearchRequest.Size != null) query = query.Where(ws => ws.WhaleSpecies.Size == whaleSightingSearchRequest.Size).ToList();
+            
+                if (whaleSightingSearchRequest.MaxLongitude != null) query = query.Where(ws => ws.LocationLongitude <= whaleSightingSearchRequest.MaxLongitude).ToList();
+
+                if (whaleSightingSearchRequest.MinLongitude != null) query = query.Where(ws => ws.LocationLongitude >= whaleSightingSearchRequest.MinLongitude).ToList();
+
+                if (whaleSightingSearchRequest.MaxLatitude != null) query = query.Where(ws => ws.LocationLatitude <= whaleSightingSearchRequest.MaxLatitude).ToList();
+
+                if (whaleSightingSearchRequest.MinLatitude != null) query = query.Where(ws => ws.LocationLatitude >= whaleSightingSearchRequest.MinLatitude).ToList();
```

Comment:
Can we add these as multiple `.Where()` statements to one query?
```suggestion
                query = query
                    .Where(ws => string.IsNullOrEmpty(whaleSightingSearchRequest.WhaleSpecies) || ws.WhaleSpecies.Name.ToLower() == whaleSightingSearchRequest.WhaleSpecies.ToLower())
                   .Where(ws => string.IsNullOrEmpty(whaleSightingSearchRequest.Colour) || ws.WhaleSpecies.Colour.ToLower() == whaleSightingSearchRequest.Colour.ToLower());
                   etc....
```